### PR TITLE
add worker contract

### PR DESCRIPTION
**Pull request by rcooke-warwick**
This change creates a "contract" within the worker container - a json object that is populated based on tags placed on the worker device via the cloud. This contract is exposed via a http endpoint (`/contract`). At the moment, it only has 3 fields:

```json
{
"dut": "raspberrypi3",
"screencapture": true,
"modem": false
}
```
That can be set using tags on the device of the same name. (e.g setting a tag `screencapture: true` on the worker will set that in the contract. It can be extended by adding to the `supportedTags` array in `worker/lib/index.ts

##################

Within a test suite, a `workerContract` object can be created:

```js
  // Get worker setup info
  this.suite.context.set({
	  workerContract: await this.context.get().worker.getContract()
  })
```

And then at the top of a test, you can define the schema to use to evaluate whether or not to run the test - for example this will restrict the test only to run if the screencapture property is `true` (signalling that the worker has a hdmi capture device set up):

```js
module.exports = {
	title: 'Balena boot splash tests',
	workerContract: {
		type: 'object',
		required: ['screencapture'],
		properties: {
			screencapture: {
				type: 'boolean',
				const: true
			},
		},
	},
	tests: [
		{
			title: 'Reboot test',
			run: async function(test) { /* test function in here */ }
		},
	],
};

```

Change-type: patch
Signed-off-by: Ryan Cooke <ryan@balena.io>

### CODE REVIEW

File: worker/lib/index.ts [hunk 1]
```diff
@@ -44,6 +45,34 @@ async function setup(): Promise<express.Application> {
...
+	}
+	const balena = getSdk({
+		apiUrl: "https://api.balena-cloud.com/"
+	});	
+	if(typeof process.env.BALENA_API_KEY === 'string' && typeof process.env.BALENA_DEVICE_UUID === 'string'){
+		await balena.auth.loginWithToken(process.env.BALENA_API_KEY);
+		const tags = await balena.models.device.tags.getAllByDevice(process.env.BALENA_DEVICE_UUID);
+		for(let tag of tags){
+			if(supportedTags.includes(tag.tag_key)){
+				if(tag.value === "true")
```
Comment (klutchell) [inline worker/lib/index.ts:66]: We could also assume the presence of a tag with no value to imply true?
```suggestion
				if (!tag.value || tag.value === "true")
```
Need to verify what the sdk returns when a tag has no value, I've made some assumptions here.

**Comment by klutchell** [comment]
Looks pretty good to me, does it run? It looks like it's a non-breaking change?

**Comment by rcooke-warwick** [comment]
@balena-ci retest

**Comment by rcooke-warwick** [comment]
@balena-ci rebase

**Comment by rcooke-warwick** [comment]
@balena-ci I self-certify!

**Pull request merged**